packages/react-charting/src/utilities/utilities.ts

```diff
@@ -132,7 +137,55 @@ export interface IYAxisParams {
  * @export
  * @param {IXAxisParams} xAxisParams
  */
-export function createNumericXAxis(xAxisParams: IXAxisParams, culture?: string) {
+export function createNumericXAxis(xAxisParams: IXAxisParams, chartType: ChartTypes, culture?: string) {
+  switch (chartType) {
+    case ChartTypes.HorizontalBarChartWithAxis:
+      return createNumericXAxisForHBCWA(xAxisParams, culture);
+    default:
+      return createNumericXAxisForOtherCharts(xAxisParams, culture);
+  }
+}
+
+/**
+ * Create Numeric X axis for Horizontal Bar Chart With Axis
+ * @export
+ * @param {IXAxisParams} xAxisParams
+ */
+export function createNumericXAxisForHBCWA(xAxisParams: IXAxisParams, culture?: string) {
+  const {
+    domainNRangeValues,
+    showRoundOffXTickValues = false,
+    xAxistickSize = 6,
+    tickPadding = 10,
+    xAxisCount = 6,
+    xAxisElement,
+  } = xAxisParams;
+  const xAxisScale = d3ScaleLinear()
+    .domain([domainNRangeValues.dStartValue, domainNRangeValues.dEndValue])
+    .range([domainNRangeValues.rStartValue, domainNRangeValues.rEndValue]);
+  showRoundOffXTickValues && xAxisScale.nice();
+
+  const xAxis = d3AxisBottom(xAxisScale)
+    .tickSize(xAxistickSize)
+    .tickPadding(tickPadding)
+    .ticks(xAxisCount)
+    .tickSizeInner(-(xAxisParams.containerHeight - xAxisParams.margins.top!))
```

Comment:
I see only addition ofr tickSizeInner. Can a single function handle for both HBCWA and other cases

Reply:
Done